spec.bs

```diff
@@ -464,18 +499,27 @@ members:
 <dl dfn-type="dict-member" dfn-for="SecurePaymentConfirmationRequest">
     :  <dfn>challenge</dfn> member
     :: A random challenge that the relying party generates on the server side to prevent replay attacks.
+    
+    **TODO**: Clarify if this challenge should be set to 32 bytes as for WebAuthn, or can be any string, and the Browser agent will hash it.
 
     :  <dfn>credentialIds</dfn> member
-    :: The list of credential identifiers for the given instrument.
+    :: The list of webauthn credential identifiers for the given instrument.
 
     :  <dfn>instrument</dfn> member
     :: The description of the instrument name and icon to display during enrollment and to be signed along with the transaction details.
 
     :  <dfn>timeout</dfn> member
     :: The number of milliseconds before the request to sign the transaction details times out. At most 1 hour.
+    
+    **TODO**: We need to define how this timeout is calculated. E.g. If the timeout is 60 seconds, and the uer spends 25 seconds on the SPC payment page in the browser, does the WebAuthn request then get initialized with 35 seconds or with 60 seconds? 
+    
+    **TODO**: What do we do if the SPC page is open for more than 60 seconds? Do we cancel without attempting WebAuthn?
 
     :  <dfn>payeeOrigin</dfn> member
     :: The origin of the payee that this SPC call is for (e.g., the merchant).
+    
+    **TODO**: Will this be validated in any way? 
```

Comment:
It is impossible for the UA to validate this in the redirect case. This is identical to displayName and icon, where we are just relaying information from the SPC caller to the user, and then signing it to tell the RP that this is what we showed to the user. It's up to the RP to decide if it trusts the SPC caller to be doing a transaction for the payeeOrigin.

Reply:
Thanks.